File: front/src/models/Event.ts
import type {Attendee, AttendeeState} from "@/models/Attendee.ts";
import {DateHelper} from "@/helpers/DateHelper.ts";

export class Event {

  private name: string;
  private token: string;
  private attendances: Map<number, Attendee[]>;

  public constructor(name: string, token: string, dates : Map<number, Attendee[]>) {
    this.name = name;
    this.token = token;
    this.attendances = dates;
  }

  public getName() : string {
    return this.name;
  }

  public getToken() : string {
    return this.token;
  }

  public getDates() : number[] {
    return Array.from(this.attendances.keys()).sort();
  }

  public getAttendees(date : number) : Attendee[] {
    if(this.attendances.has(date)){
      return this.attendances.get(date)!; //Fuck TS
    }
    return [];
  }

  public getMaxAttendees(){
    let max = 0;
    this.attendances.forEach(attendee => {
      if(attendee.length > max) max = attendee.length;
    })
    return max;
  }

}

export interface EventState {
  name : string
  token : string
  dates: Map<number, AttendeeState[]>;
}

File: back/src/main/java/be/naaturel/letsmeet/core/models/Event.java
package be.naaturel.letsmeet.core.models;

import java.util.Set;

public class Event {

    private String name;

    private Set<Attendance> attendances;

    public Event(String name, Set<Attendance> attendances) {
        this.name = name;
        this.attendances = attendances;
    }

    public String getName() {
        return name;
    }

    public Set<Attendance> getAttendances() {
        return this.attendances;
    }

}
